scriptworker/cot/verify.py

```diff
@@ -1046,9 +1100,94 @@ def _get_additional_github_releases_jsone_context(parent_link, decision_link):
             'release': {
                 'tag_name': get_revision(task, source_env_prefix),
                 'target_commitish': get_branch(task, source_env_prefix),
+                'published_at': get_push_date_time(task, source_env_prefix),
             },
             'sender': {
-                'login': get_triggered_by(task, source_env_prefix),
+                'login': user,
+            },
+        },
+    }
+
+
+async def _get_additional_github_pull_request_jsone_context(decision_link):
```

Comment:
Overall, awesome that we're supporting pull requests; that will help a great deal. We do need to be a bit careful about permissions.

In a pull request, which `.taskcluster.yml` do we use to generate the tasks? The pull request's `.taskcluster.yml`? We can't necessarily trust the code or config in the pull request, especially since this is pre-review from an arbitrary user. We may want something like `is_try` to notice that our `tasks_for` specifies a pull request.

Reply:
Yeah, we need to rely on the PR's `.taskcluster.yml`. That's useful to test out staging releases. I like the idea of `is_try`. I just implemented something like this!

Reply:
I don't think we can trust `tasks_for` to determine if something is a pull request, since it comes from the untrusted `.taskcluster.yml`.

Reply:
I agree. This can easily be hardcoded. Just like `is_try`, I implemented a couple checks on the URL used in the decision task https://github.com/mozilla-releng/scriptworker/pull/307/files#diff-27eeb4b1649924d24d3d4b07fc7da4d5R448. These can be changed too, but if `metadata.source` is forged, then `.taskcluster.yml` must match something that was shipped on the official repository. This makes the attack surface lower, IMO.